src/http-gateways/trustless-gateway.md

```diff
@@ -301,6 +302,112 @@ of their presence in the DAG or the value assigned to the "dups" parameter, as
 the raw data is already present in the parent block that links to the identity
 CID.
 
+## `meta` (content type parameter)
+
+The `meta` parameter allows clients to request the server to include additional metadata about the CAR along with the response body.
+
+The value of this parameter includes both the location where the metadata is given (e.g. `eof`) as well as the type of data received (e.g. `json`) separated by a `+`, to give a value such as `meta=eof+json`
+
+When the location parameter is set to `eof`, which is currently the only supported value, the server SHOULD respond with the following response body:
+
+```
+<Response body as CARv1 stream> <0x00 byte> <Metadata>
+```
+
+The only supported value for the data type parameter is `json`. This signifies that the metadata MUST be a JSON object.
+
+This parameter MUST only be used with CAR `version=1`.
+
+When the parameter is not set or does not equal `eof+json`, the server SHOULD not add any extra blocks to the response, neither the 0x00 byte nor any metadata.
+
+When `meta=eof+json`, the JSON object SHOULD conform to the following [JSON schema](https://json-schema.org/).
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "data": {
+      "type": "object",
+      "description": "Properties of the response"
+    },
```

Comment:
Discussion point:

In the current proposal, the top-level "data" object combines fields about "what was requested" (e.g. CAR & DAG params) with "what was returned" (e.g. CARv1 length in bytes).

I'd like to discuss an alternative: split `data` into two fields `req` and `res`. The first will describe what the client requested, the second will describe what the server returned.

Such division would allow us to shorten field names, e.g. `data.car_params.dup` can become `req.dups`.

Reply:
Splitting into `req` and `res` sgtm, improves clarity